technical.md

```diff
@@ -5,100 +5,212 @@ title: OpenMath, A Technical Overview
+
+The binary encoding mostly replaces the spacious tags of XML with single bytes, and packs identifiers and numerical data tight. A line by line conversion of the above to binary, writing bytes either in hexadecimal or as `"string"` (meaning the UTF-8 encoding of the quoted characters), is with spaces inserted for clarity just
+```
+  1a
+    080406 "fns1" "lambda"
+    1c 0501 "x" 1d
+    10 080604 "arith1" "plus"
+      10 080605 "arith1" "power" 0501 "x" 0105 11
+      10 080605 "arith1" "times" 
+        01fd
+        10 080605 "arith1" "power" 0501 "x" 0104 11
+      11
+      10 080605 "arith1" "times" 
+        10 080508 "nums1" "rational"
+          0103 010e
+        11
+        10 080605 "arith1" "power" 0501 "x" 0102 11
+      11
+      01f8
+    11
+  1b
+```
+Now the most spacious parts are the symbol names, due to not playing favourites. Even these may be further abbreviated by using _references_ to repeated subobjects:
+```
+  1a
+    080406 "fns1" "lambda"
+    1c 8501 "x" 1d
+    10 080604 "arith1" "plus"
+      10 880605 "arith1" "power" 1e00 0105 11
+      10 880605 "arith1" "times" 
+        01fd
+        10 1e01 1e00 0104 11
+      11
+      10 1e02 
+        10 080508 "nums1" "rational"
+          0103 010e
+        11
+        10 1e01 1e00 0102 11
+      11
+      01f8
+    11
+  1b
+```
+Here we are down to 61 bytes for the structure and another 56 bytes for names (from 506 significant characters in the XML encoding) — certainly still longer than `x->x^5-3*x^4+3/14*x^2-7`, although a fairer comparison would be with the fully parenthesised `x->((x^5)+((-3)*(x^4))+((3/14)*(x^2))+(-7))` since OpenMath makes no presumptions about operation priorities.
 
+The third standardised OpenMath encoding is (strict) Content MathML (cMML). This is a sibling of the more well-known _Presentation_ MathML (pMML) that is sometimes used for math formulae in web settings, but as the name says encoding _content_ (semantics) rather than _presentation_ (looks). Being another XML encoding, it is in most ways similar to OpenMath-XML, but the tag names are different, and cMML prefers to put names in element contents rather than attributes, so our initial `<OMS cd="transc1" name="sin"/>` example becomes
 ```XML
-<OMOBJ>
-  <OME>
-    <OMS name="unexpected_symbol" cd="error1">
-    <OMS name="sine" cd="transc1">
-  </OME>
-</OMOBJ>
+  <csymbol cd="transc1">sin</csymbol>
 ```
+in cMML.
 
-represents the error which might be generated when an application sees a symbol it doesn't recognise from a CD it thought it knew about.
+Historically there also was an encoding called _prefix_ or _functional_, with a syntax based on LISP. This was never codified in the standard, but the content dictionary pages on the OpenMath website still offers showing OpenMath objects in prefix notation.
```

Comment:
Drop this paragraph, the prefix form is something I invented to make a human readable/writable encoding to go into the html presentation (the rendered mathml possibility came later) the original lisp encoding was different as was really list so (sin theta) not sin(theta) as in the prefix rendering.